op-node/rollup/derive/l1_block_info.go

```diff
@@ -159,7 +164,17 @@ func (info *L1BlockInfo) unmarshalBinaryBedrock(data []byte) error {
 // | 32      | BlockHash                |
 // | 32      | BatcherHash              |
 // +---------+--------------------------+
+func (info *L1BlockInfo) marshalBinaryIsthmus() ([]byte, error) {
+	// Isthmus is the same as Ecotone
+	out, err := info.marshalBinaryEcotone()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal Ecotone l1 block info: %w", err)
+	}
+	// replace the first 4 bytes of the signature with L1InfoFuncIsthmusBytes4
+	copy(out[:4], L1InfoFuncIsthmusBytes4) // CHECK: is this correct? or is it too nasty?
```

Comment:
I don't think we want to do this altho it is valid, a couple of approaches:
- copy/paste code
- modularize required code into a free function and use in both `marshalBinaryIsthmus` and `marshalBinaryEcotone`